plugin/pki/util.go

```diff
@@ -272,46 +278,49 @@ func updateAccessToken(b *backend, ctx context.Context, req *logical.Request, cf
 
 	tppConnector.SetHTTPClient(httpClient)
 
-	b.Logger().Debug("Refreshing token")
+	b.Logger().Info("Refreshing access_token")
 	var resp tpp.OauthRefreshAccessTokenResponse
 	resp, err = tppConnector.RefreshAccessToken(&endpoint.Authentication{
 		RefreshToken: refreshToken,
 		ClientId:     "hashicorp-vault-by-venafi",
 		Scope:        "certificate:manage,revoke",
 	})
 	if resp.Access_token != "" && resp.Refresh_token != "" {
-		err = storeAccessData(b, ctx, req, role.Name, resp)
+		b.Logger().Info("Storing new token")
+		err = storeAccessData(b, ctx, req, role, resp)
 	}
 	return err
 }
 
-func storeAccessData(b *backend, ctx context.Context, req *logical.Request, roleName string, resp tpp.OauthRefreshAccessTokenResponse) error {
-	entry, err := b.getRole(ctx, req.Storage, roleName)
+func storeAccessData(b *backend, ctx context.Context, req *logical.Request, role *roleEntry, resp tpp.OauthRefreshAccessTokenResponse) error {
```

Comment:
I see you are changing the function signature... Maybe this is a part of other Code Review because is out of the objective of this CR.

Reply:
It is part of this PR as we intended to solve every bug related in the token refresh process. As you can see, I changed it as we no longer needed to retrieve the role back again since we already have it during the process.

Reply:
got it